src/app.js

```diff
@@ -0,0 +1,314 @@
+  switch (mode) {
+    case 'eng':
+      mode = 'ru';
+      localStorage.setItem('mode', 'ru');
+      changeKeyboard(ru);
+      break;
+    case 'engCaps':
+      mode = 'ruCaps';
+      localStorage.setItem('mode', 'ruCaps');
+      changeKeyboard(ruCaps);
+      break;
+    case 'ruCaps':
+      mode = 'engCaps';
+      localStorage.setItem('mode', 'engCaps');
+      changeKeyboard(engCaps);
+      break;
+    case 'ru':
+      mode = 'eng';
+      localStorage.setItem('mode', 'eng');
+      changeKeyboard(eng);
+      break;
+    default:
+      break;
+  }
+};
+
+const handleFuncButton = (code) => {
+  const textarea = document.getElementsByTagName('textarea')[0];
+  switch (code) {
+    case 'Backspace':
+      textarea.value = textarea.value.slice(0, textarea.value.length - 1);
+      break;
+    case 'CapsLock':
+      document.getElementsByClassName('button')[29].classList.toggle('clicked');
+      changeCapsLock();
+      break;
+    case 'Tab':
+      textarea.value += '\t';
+      break;
+    case 'Delete':
+      textarea.value = textarea.value.slice(0, textarea.selectionStart)
+        .concat(textarea.value.slice(textarea.selectionEnd, textarea.value.length));
+      break;
+    case 'Del':
+      textarea.value = textarea.value.slice(0, textarea.selectionStart)
+        .concat(textarea.value.slice(textarea.selectionEnd, textarea.value.length));
+      break;
+    case 'Enter':
+      textarea.value += '\n';
+      break;
+    case 'ShiftLeft' || 'ShiftRight' || 'ControlLeft' || 'MetaLeft' || 'AltLeft' || 'AltRight' || 'ControlRight':
+      break;
+    default:
+      break;
+  }
+};
+
+const newEvent = (element) => {
+  const ev = new Event('mousedown');
+  element.dispatchEvent(ev);
+  ev.stopPropagation();
+};
+
+const keydown = (event) => {
+  const index = evCodeArr.indexOf(event.code);
+  if (event.shiftKey && event.altKey) {
+    changeLang();
+  } else if (event.code === 'CapsLock' || event.code === 'Backspace' || event.code === 'Tab' || event.code === 'Delete' || event.code === 'Enter' || event.code === 'ShiftLeft' || event.code === 'ShiftRight' || event.code === 'ControlLeft' || event.code === 'MetaLeft' || event.code === 'AltLeft' || event.code === 'AltRight' || event.code === 'ControlRight') {
+    const element = document.getElementsByClassName('button')[index];
+    newEvent(element);
+    handleFuncButton(event.code);
+  } else if (index >= 0) {
+    const element = document.getElementsByClassName('button')[index];
+    document.getElementsByTagName('textarea')[0].value += element.textContent;
+    newEvent(element);
+  }
+};
+
+const keyup = (event) => {
```

Comment:
Pay attention to this function.
Sometimes it not works for "Ctrl" "Backspace" buttons

Reply:
Fixed.